File: Core/Shared/JavaLib/src/net/lukx/jchatter/lib/repos/AbstractRepo.java
package net.lukx.jchatter.lib.repos;

import net.lukx.jchatter.lib.PublicApi;
import net.lukx.jchatter.lib.comms.Communicable;
import net.lukx.jchatter.lib.comms.LoginHeader;

import java.util.HashMap;

public abstract class AbstractRepo {

    /***
     * @return the controller this class send data to
     */
    protected abstract String getController();

    @PublicApi
    protected Communicable communicable;

    @PublicApi
    protected LoginHeader loginHeader = new LoginHeader();

    AbstractRepo(Communicable communicable){
        this.communicable = communicable;
    }

    @PublicApi
    protected class IDObject {
        @PublicApi
        private String Login = getLoginHeader().getLogin();
        @PublicApi
        private String Password = getLoginHeader().getPassword();
        @PublicApi
        private int ID;

        IDObject(int id){
            this.ID = id;
        }
    }

    @PublicApi
    public class KeyValuePair {
        String key;
        Object value;

        @PublicApi
        public KeyValuePair(){

        }

        KeyValuePair(String key, Object value){
            this.key = key;
            this.value = value;
        }
    }

    @PublicApi
    protected Object createCustomObjectWithHeader(KeyValuePair... keyValuePair){
        @SuppressWarnings("unchecked")
        HashMap<String,Object> hm = (HashMap<String,Object>)createCustomObject(keyValuePair);
        hm.put("Login",getLoginHeader().getLogin());
        hm.put("Password",getLoginHeader().getPassword());
        return hm;
    }

    @PublicApi
    protected Object createCustomObject(KeyValuePair... keyValuePair){
        HashMap<String,Object> map = new HashMap<>();
        for (KeyValuePair kv : keyValuePair)
        {
            map.put(kv.key,kv.value);
        }
        return map;
    }

    @PublicApi
    protected Object createIdObject(int id){
        return new IDObject(id);
    }

    @PublicApi
    public LoginHeader getLoginHeader() {
        return loginHeader;
    }
}

File: Core/Wpf/ChatterWpf/ChatterWpf/Com/repos/AbstractRepo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Collections.Generic.Dictionary;

public abstract class AbstractRepo
{
    protected abstract string getController();

    protected ICommunicable communicable;

    private LoginHeader loginHeader = new LoginHeader();

   

    AbstractRepo(ICommunicable communicable)
    {
        this.communicable = communicable;
    }

    private class IDObject
    {
        private string Login = getLoginHeader().getLogin();
        private string Password = getLoginHeader().getPassword();
        private int ID;

        IDObject(int id)
        {
            this.ID = id;
        }
    }

    public class KeyValuePair
    {
        string key;
        Object value;

        public KeyValuePair()
        {

        }

        KeyValuePair(string key, Object value)
        {
            this.key = key;
            this.value = value;
        }
    }
    //KeyValuePair...keyValuePair ?
    protected Object createCustomObjectWithHeader(KeyValuePair keyValuePair)
    {
        Dictionary<String, Object> hm = (Dictionary<String, Object>)createCustomObject(keyValuePair);
        hm[Login] = getLoginHeader().getLogin();
        hm[Password] = getLoginHeader().getPassword();
        return hm;
    }

    protected Object createCustomObject(KeyValuePair keyValuePair)
    {
        Dictionary<String, Object> map = new Dictionary<>();
        foreach(KeyValuePair kv in keyValuePair)
        {
            map[kv.key] = kv.value;
        }
        return map;
    }



    Object createIdObject(int id)
    {
        return new IDObject(id);
    }

    LoginHeader getLoginHeader()
    {
        return loginHeader;
    }
}
